### PR TITLE
[IO-744] FileWriterWithEncoding for an existing file no longer truncates the file.

### PR DESCRIPTION
**Pull request by rbri**

### CODE REVIEW

**Comment by garydgregory** [comment]
@rbri 
Thank you for your PR. See comments.

File: src/main/java/org/apache/commons/io/output/FileWriterWithEncoding.java [hunk 1]
```diff
@@ -227,7 +227,12 @@ private static Writer initWriter(final File file, final Object encoding, final b
         OutputStream stream = null;
         final boolean fileExistedAlready = file.exists();
         try {
-            stream = Files.newOutputStream(file.toPath(),  append ? StandardOpenOption.APPEND : StandardOpenOption.CREATE);
+            if (append) {
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/io/output/FileWriterWithEncoding.java:230]: Not sure why the fix has to be exploded into so much code when all you need is an additional argument to the JRE API. Please see git master for a much simpler fix.

File: src/test/java/org/apache/commons/io/output/FileWriterWithEncodingTest.java [hunk 2]
```diff
@@ -197,6 +198,25 @@ public void constructor_File_nullFile() {
         assertFalse(file1.exists());
     }
 
+    @Test
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/io/output/FileWriterWithEncodingTest.java:201]: This test is now in git master, TY. You are credited in `changes.xml`.

File: src/main/java/org/apache/commons/io/output/FileWriterWithEncoding.java (hunk 1 above)
Comment (rbri) [reply src/main/java/org/apache/commons/io/output/FileWriterWithEncoding.java:230]: @garydgregory Thanks for the warm welcome as someone trying to help.
Have done another PR that shows, that your optimized version of the fix now breaks the ctor with the append option set to true.

**Pull request closed without merging**